domains/transfer/Transfer.API/API.IntegrationTests/Transfer/TransferAgreementCleanup/TransferAgreementCleanupTests.cs

```diff
@@ -0,0 +1,135 @@
+        var dbContext = scope.ServiceProvider.GetRequiredService<ApplicationDbContext>();
+
+        dbContext.TransferAgreements.RemoveRange(dbContext.TransferAgreements);
+        await dbContext.SaveChangesAsync();
+
+        var expiredTa = new TransferAgreement
+        {
+            Id = Guid.NewGuid(),
+            EndDate = DateTimeOffset.UtcNow.AddHours(-1),
+            ReceiverTin = "12345678",
+            SenderName = "SomeSender",
+            SenderTin = tin,
+            ReceiverReference = Guid.NewGuid(),
+            SenderId = sub,
+            StartDate = DateTimeOffset.UtcNow.AddDays(-1),
+            TransferAgreementNumber = 0
+        };
+        var nullEndDateTa = new TransferAgreement
+        {
+            Id = Guid.NewGuid(),
+            EndDate = null,
+            ReceiverTin = "12345679",
+            SenderName = "SomeSender",
+            SenderTin = tin,
+            ReceiverReference = Guid.NewGuid(),
+            SenderId = sub,
+            StartDate = DateTimeOffset.UtcNow.AddDays(-1),
+            TransferAgreementNumber = 1
+        };
+        var endDateTa = new TransferAgreement
+        {
+            Id = Guid.NewGuid(),
+            EndDate = DateTimeOffset.UtcNow.AddHours(1),
+            ReceiverTin = "12345677",
+            SenderName = "SomeSender",
+            SenderTin = tin,
+            ReceiverReference = Guid.NewGuid(),
+            SenderId = sub,
+            StartDate = DateTimeOffset.UtcNow.AddDays(-1),
+            TransferAgreementNumber = 2
+        };
+
+        dbContext.TransferAgreements.Add(expiredTa);
+        dbContext.TransferAgreements.Add(nullEndDateTa);
+        dbContext.TransferAgreements.Add(endDateTa);
+        await dbContext.SaveChangesAsync();
+
+        var tas = await dbContext.RepeatedlyQueryUntilCountIsMet<TransferAgreement>(2);
+
+        tas.Count.Should().Be(2);
+        tas.Select(x => x.Id).Should().Contain(nullEndDateTa.Id);
+        tas.Select(x => x.Id).Should().Contain(endDateTa.Id);
+        tas.Select(x => x.Id).Should().NotContain(expiredTa.Id);
+    }
+
+    [Fact]
+    public async Task ShouldProduceActivityLogEntriesForReceiverAndSender()
+    {
+        var receiverTin = "12345677";
+        using var scope = factory.Services.CreateScope();
+        var dbContext = scope.ServiceProvider.GetRequiredService<ApplicationDbContext>();
+
+        dbContext.TransferAgreements.RemoveRange(dbContext.TransferAgreements);
+        await dbContext.SaveChangesAsync();
+
+        var expiredTa = new TransferAgreement
+        {
+            Id = Guid.NewGuid(),
+            EndDate = DateTimeOffset.UtcNow.AddHours(-1),
+            ReceiverTin = receiverTin,
+            SenderName = "SomeSender",
+            SenderTin = tin,
+            ReceiverReference = Guid.NewGuid(),
+            SenderId = sub,
+            StartDate = DateTimeOffset.UtcNow.AddDays(-1),
+            TransferAgreementNumber = 0
+        };
```

Comment:
Is this a valid TA? start date must be before enddate

Reply:
StartDate is before EndDate? It says AddDays :)

Reply:
ooopppppps